ios/brave-ios/Sources/Brave/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift

```diff
@@ -43,9 +43,35 @@ class TabLocationView: UIView {
   let contentView = UIView()
   private var tabObservers: TabObservers!
   private var privateModeCancellable: AnyCancellable?
+  private var _url: URL?
 
   var url: URL? {
-    didSet {
+    get {
+      _url
+    }
+
```

Comment:
I have 2 problems with this in particular:
1. We're now altering the URL's passed down from above, meaning if its read back out its not the same as what was assigned. We probably want to change the name at this point then to be more obvious that what you're reading is something new, e.g. `displayURL` or similar
2. If we need to compute additional logic now for every URL set we should probably just make the original property have a private setter instead, and introduce a method to handle this rather than have a proxy. Semantically this makes more sense

e.g.

```swift
private(set) var displayURL: URL?
func setURL(_ url: URL) { ... }
```

Reply:
Got rid of all of it and put it in an extension. Now I call it whenever the URL will be displayed, rather than altering the URL itself.